tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_Characteristics.cpp

```diff
@@ -127,16 +133,105 @@ void test_with_normal_along_coordinate_axes(
           nn_generator, nn_distribution, used_for_size));
 }
 
+template <size_t Dim>
+void test_characteristic_matrices(const double used_for_size) noexcept {
+  MAKE_GENERATOR(generator);
+  const auto spatial_metric = hydro::TestHelpers::random_spatial_metric<Dim>(
+      make_not_null(&generator), used_for_size);
+  const auto det_and_inv_metric = determinant_and_inverse(spatial_metric);
+  const auto lorentz_factor = hydro::TestHelpers::random_lorentz_factor(
+      make_not_null(&generator), used_for_size);
+  const auto spatial_velocity = hydro::TestHelpers::random_velocity(
+      make_not_null(&generator), lorentz_factor, spatial_metric);
+
+  // Make consistent set of thermodynamic variables, along with their
+  // derived quantities for a given equation of state.
+  const EquationsOfState::IdealFluid<true> equation_of_state{5.0 / 3.0};
```

Comment:
I recommend testing with a Polytrope EoS too.

For this equation of state `kappa` goes to zero, so it's good to have a test that checks the choice of left/right vectors has no divisions by kappa. I did this wrong in the Newtonian case, and am currently working to fix it...

Reply:
Actually, in the relativistic case kappa = rho c_s^2 in this case (rho = rest mass). I think the difference is because of including rest mass in the energy density. The eigenvectors should be fine, the way I wrote them. It would be good to check numerically!

Reply:
I haven't looked into this yet, but our `PolytropicFluid` equation of state sets `kappa_times_p_over_rho_squared` to 0 whether relativistic or not.